### PR TITLE
Add test-filter to Config

### PR DESCRIPTION
**Pull request by CharlesD91**
#### What it does

Adds `test-filter` to `Config`

#### How it works

It restricts the tests that are run for every mutation, so that we can ignore heavy `integration` tests and focus on `unit` ones

### CODE REVIEW

File: core/src/test/scala/stryker4s/config/TestFilterTest.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package stryker4s.config
+
+import stryker4s.testutil.Stryker4sSuite
+
+class TestFilterTest extends Stryker4sSuite {
+  describe("filter with default") {
+    implicit val conf: Config = Config.default
+    it("should work") {
```
Comment (hugo-vrijswijk) [inline core/src/test/scala/stryker4s/config/TestFilterTest.scala:8]: The 'should work' is a little duplicate three times. Perhaps a single `describe` of the function you are testing, with an `it` of the changed system under test description would be clearer?

Comment (CharlesD91) [reply core/src/test/scala/stryker4s/config/TestFilterTest.scala]: I fixed this

**Comment by CharlesD91** [comment]
> Thank you for your PR! Overall the code looks good, I just have a small comment. Could you also add some documentation about the new config option?

I updated the doc. Also I did a small change on how the `TestFilter` is used in the `SbtMutantRunner` settings: if the `test-filter` wasn't set in the config, then nothing related to it will happen in the code (better sandboxing of the different logics in case of a bug)

**Comment by cbrunnkvist** [comment]
I was just about to open a bug report for v0.7.3 about this not working correctly, when I by chance discovered this PR and that it was closed less than a week ago. 👍 

I had perhaps naively assumed that the `test-filter` option was already in all recent releases. Perhaps this means it is time to start updating _"added in version..."_ to the CONFIGURATION.md items. 😓

**Comment by hugo-vrijswijk** [comment]
@cbrunnkvist Excellent idea! Would be helpful for a lot of people. In general, our documentation always reflects the latest version. But feel free to open a PR!

**Pull request merged**